### PR TITLE
Fix plain encoding

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/src/io/object_store.rs [hunk 1]
```diff
@@ -167,4 +167,9 @@ impl ObjectStore {
             Err(e) => Err(e.into()),
         }
     }
+
+    /// Get file size.
+    pub async fn size(&self, path: &Path) -> Result<usize> {
+        Ok(self.inner.head(path).await?.size)
```
Comment (changhiskhan) [inline rust/src/io/object_store.rs:173]: might be slow if it's s3 path and this gets called a bunch of times? ok for now as it's a utility for testing. add a todo here or something?

File: rust/src/encodings/plain.rs [hunk 2]
```diff
@@ -99,9 +98,13 @@ impl<'a> PlainEncoder<'a> {
             self.encode_boolean(boolean_arr).await?;
         } else {
             let byte_width = array.data_type().byte_width();
-            self.writer
-                .write_all(&data.buffers()[0].as_slice()[0..array.len() * byte_width])
-                .await?;
+            let slice = unsafe {
+                from_raw_parts(
```
Comment (changhiskhan) [inline rust/src/encodings/plain.rs:102]: does this need to be done for boolean as well?

File: rust/src/encodings/plain.rs [hunk 3]
```diff
@@ -99,9 +98,13 @@ impl<'a> PlainEncoder<'a> {
...
+                    data.buffers()[0].as_ptr().add(data.offset() * byte_width),
+                    data.len() * byte_width,
```
Comment (changhiskhan) [inline rust/src/encodings/plain.rs:104]: what's the diff between array.len() vs data.len() ?

File: rust/src/encodings/plain.rs (hunk 2 above)
Comment (eddyxu) [reply rust/src/encodings/plain.rs:102]: Boolean needs to shift to the byte array,thus have to copy the data to new buffer, so this does not work for boolean.

File: rust/src/encodings/plain.rs (hunk 3 above)
Comment (eddyxu) [reply rust/src/encodings/plain.rs:104]: Oh no real difference, was looking at 

https://docs.rs/arrow-array/33.0.0/src/arrow_array/array/primitive_array.rs.html#308-311

File: rust/src/io/object_store.rs (hunk 1 above)
Comment (eddyxu) [reply rust/src/io/object_store.rs:173]: i am not sure what optimization we can do here. And getting metadata on s3 is slow is well-understood by users, they should have the expectation?

Unlikely the upstream object store crate will explicitly call out that operations on S3 is slow but on local filesystem is faster?

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**